lib/spack/spack/platforms/cray.py

```diff
@@ -119,22 +131,66 @@ def _default_target_from_env(self):
         if getattr(self, 'default', None) is None:
             bash = Executable('/bin/bash')
             output = bash(
-                '-lc', 'echo $CRAY_CPU_TARGET',
+                '--norc', '--noprofile', '-lc', 'echo $CRAY_CPU_TARGET',
```

Comment:
@becker33 I am a bit late with this but what output do you expect from this command? The environment is clean and `bash` is called in the login mode with `--noprofile`. Doesn't this mean that the output is always empty?